### ISSUE TITLE
CopyWith method generated on fields in fragments expect functions rather than values themselves

### ISSUE THREAD

**Issue by Hari-07** [heftapp/graphql_codegen#104]
Suppose a schema and fragments as given below

```graphql
type Tournament {
	event: String!
	match: Match
}

type Match {
	homeTeam: Score
	awayTeam: Score
}

type Score {
	teamName: String!
	value: Int!
}
```

```graphql
fragment Tournament on Tournament {
	event
	match {
		homeTeam {
			...Score
		}
		awayTeam {
			...Score
		}
	}
}

fragment Score on Score {
	teamName
	value
}
```

The code generated creates a class called FragmentTournament$match whose copywith is as given below
```dart
extension UtilityExtensionFragmentTournament$match on FragmentTournament$match {
  FragmentTournament$match copyWith(
          {FragmentTournament$match$homeTeam? Function()? homeTeam,
          FragmentTournament$match$awayTeam? Function()? awayTeam,
          String? $__typename}) =>
      FragmentTournament$match(
          homeTeam: homeTeam == null ? this.homeTeam : homeTeam(),
          awayTeam: awayTeam == null ? this.awayTeam : awayTeam(),
          $__typename: $__typename == null ? this.$__typename : $__typename);
}
``` 
It's expected to take in the values rather than functions that return those values, as is the norm in copyWith methods

**Comment by budde377** [comment]
Alright, how would you express these two cases with your approach:

* copy with overwriting a nullable value with `null`?
* copy with preserving a nullable value?

**Comment by budde377** [comment]
Thank you for sharing this! It's definitely a good source of improvement and I like the deep copy functionality.

**Comment by Hari-07** [comment]
Rather than reinvent the wheel, it might be easier for you to create freezed classes from the graphql and then let it handle the utility methods on it. It'll add a dependency, but I think it's worth it

**Comment by budde377** [comment]
I think that's a trade off. Generally, I'd like to reduce the number of runtime dependencies and get more control over the generated output, moving to freezed is not in line with this. Furthermore, we're in a unique position to implement this our selves in a more efficient way, because we have a very limited domain to cover. We have full control over the classes and can make anything happen. 

Lastly, this work is not super complicated, I've already implemented a better `copyWith` and have the deep copy implementation pending.

**Comment by Hari-07** [comment]
That's awesome to hear! Looking forward to your updates

**Comment by budde377** [comment]
Released new `copyWith` and deep copy in `0.10.0-beta.2`
